### PR TITLE
Make RESTEasy Reactive work with Optional temporal types

### LINKED ISSUE #28089: RESTEasy Reactive doesn't provide converters for types like `Optional<LocalDate>`
### Describe the bug

Not sure if this is expected to work or not, but while RestEasy Reactive has support for Java Time types (like `LocalDate`) out-of-the-box, this doesn't work when wrapped in an optional (e.g. `Optional<LocalDate>`).

```
2022-09-20 11:33:03,799 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#setupEndpoints threw an exception: java.lang.RuntimeException: java.lang.RuntimeException: Failed to process method 'org.acme.GreetingResource#hello'
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:313)
        at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor.setupEndpoints(ResteasyReactiveProcessor.java:584)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: Failed to process method 'org.acme.GreetingResource#hello'
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:693)
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:400)
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:284)
        ... 12 more
Caused by: java.lang.RuntimeException: Failed to find converter for java.time.LocalDate
        at org.jboss.resteasy.reactive.server.processor.generation.converters.GeneratedConverterIndexerExtension.extractConverterImpl(GeneratedConverterIndexerExtension.java:104)
        at org.jboss.resteasy.reactive.server.processor.ServerEndpointIndexer.extractConverter(ServerEndpointIndexer.java:507)
        at org.jboss.resteasy.reactive.server.processor.ServerEndpointIndexer.handleOptionalParam(ServerEndpointIndexer.java:357)
        at org.jboss.resteasy.reactive.server.processor.ServerEndpointIndexer.handleOptionalParam(ServerEndpointIndexer.java:90)
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.extractParameterInfo(EndpointIndexer.java:1251)
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:531)
        ... 14 more
```

### Expected behavior

I would have expected this to be supported. Sample code:

```java
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello(@QueryParam("when") Optional<LocalDate> when) {
        return "Hello " + when + " from RESTEasy Reactive";
    }
```

### Actual behavior

See description.

### How to Reproduce?

Unzip [code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/9606226/code-with-quarkus.zip) and run `mvn quarkus:dev`.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.12.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #28089

### CODE REVIEW

File: independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java [hunk 1]
```diff
@@ -1380,8 +1380,11 @@ protected void handleSortedSetParam(Map<String, String> existingConverters, Stri
             boolean hasRuntimeConverters, PARAM builder, String elementType) {
     }
 
-    protected void handleOptionalParam(Map<String, String> existingConverters, String errorLocation,
-            boolean hasRuntimeConverters, PARAM builder, String elementType, String genericElementType) {
+    protected void handleOptionalParam(Map<String, String> existingConverters,
+            Map<DotName, AnnotationInstance> parameterAnnotations,
+            String errorLocation,
+            boolean hasRuntimeConverters, PARAM builder, String elementType, String genericElementType,
+            MethodInfo currentMethodInfo) {
```
Comment (gsmet) [inline independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java:1387]: Is it already supported by other containers? List, Set...

Comment (geoand) [reply independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java:1387]: No it does not.

Comment (gsmet) [reply independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java:1387]: Should it? :)

Comment (geoand) [reply independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java:1387]: Likely yes, but that requires a different fix, which is probably best done in a follow up

Comment (gsmet) [reply independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java:1387]: BTW, It's an honest question, I don't know if it makes sense.

Comment (geoand) [reply independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java:1387]: Yeah, it does :). I was iniatially sceptical, but came to the conclusion that does make sense.

**Review by Sgitario — APPROVED** [review]
nice one!

**Pull request merged**